### PR TITLE
ROV-161-Home-dashboard-UI

### PR DESCRIPTION
**Pull request by Vedant-Tyagi**

### CODE REVIEW

File: src/app/application/dashboard/dashboard.component.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    top: 32%;
+    right: 0.5rem;
+    cursor: pointer;
+    background: transparent;
+    border: none;
+}
+.margin{
+    margin-left: 100px;
+}
+.font{
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.css:12]: don't declare font size here and there, use the existing font-size class from style.css

File: src/app/application/dashboard/dashboard.component.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    font-size: 29px;
+}
+.form{
+    font-size: 25px;
+}
+.no-underline{
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.css:18]: use the existing `.text-decoration-none` class for this

File: src/app/application/dashboard/dashboard.component.html [hunk 3]
```diff
@@ -1,15 +1,111 @@
...
-        <div class="col-9">
-            <h1>Dashboard</h1>
-        </div> 
-        <div class="col-3 text-end">
+        <div class="col-10">
+            <form class="form-inline my-2 my-lg-0">
+                <div class="input-group">
+                    <div class="input-group-append">
+                        <span class="input-group-text" 
+                        style="background-color:#DBDEE7;">
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:8]: don't use inline styles

File: src/app/application/dashboard/dashboard.component.html [hunk 4]
```diff
@@ -1,15 +1,111 @@
...
+                        <i class="bi bi-search btn"></i>
+                        </span>
+                    </div>
+                    <input class="form-control " style="background-color:#DBDEE7" type="search" aria-label="Search">
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:12]: don't use inline styles

Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:12]: what does `aria-label` mean

File: src/app/application/dashboard/dashboard.component.html [hunk 5]
```diff
@@ -1,15 +1,111 @@
...
+                <b class="ms-3 fw-normal">Credit Amount: </b>
+            </div>
+        </div>
+        <div class="col-3 ms-5 ">
+            <div class=" font bg-white border border-secondary  rounded-3">
+                <b class="ms-3 fw-normal " >Debit Amount: </b>
+            </div>
+        </div>
+    </div>
+    <div class="row card m-4" style=" background-color:#FEF7E5;">
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:41]: no inline-style please

File: src/app/application/dashboard/dashboard.component.html [hunk 6]
```diff
@@ -1,15 +1,111 @@
...
+                bindLabel="name" 
+                bindValue="id">
+                </ng-select>
+            </div>
+        </div>
+        <div class="col-2 ms-4">
+            <div class="form-group">
+                <label class="form-label form">Category:</label>
+                <ng-select 
+                [items]="items2"
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:95]: give proper variable name

File: src/app/application/dashboard/dashboard.component.html [hunk 7]
```diff
@@ -1,15 +1,111 @@
...
+                [minDate]="minToDate" 
+                [maxDate]="maxDate" 
+                (click)="toDatePicker.toggle()" />
+            </div>
+        </div>
+        <div class="col-2 ms-4">
+            <div class="form-group">
+                <label class="form-label form">Transaction type</label>
+                <ng-select
+                [items]="items"
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:85]: give proper variable name

File: src/app/application/dashboard/dashboard.component.html [hunk 8]
```diff
@@ -1,15 +1,111 @@
...
+                bindValue="id">
+                </ng-select>
+            </div>
+        </div>
+    <div class="row ms-5 mt-3">
+    <div class="col-3">
+        <button 
+        type="submit" 
+        class="btn btn-primary btn-green" >
+            <a href="app/consolidate-statement"
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.html:106]: this should be button with click event

File: src/app/application/dashboard/dashboard.component.ts [hunk 9]
```diff
@@ -1,18 +1,43 @@
...
+export class DashboardComponent implements OnInit {
 
-  openAddAccountModal() {
-    this.modalService.open(AddAccountComponent, { centered: true, size: 'md', });
-  }
+  items = [
+    { id: 1, name: 'All' },
+    { id: 2, name: 'Credit' },
+    { id: 3, name: 'Debit' },
+    { id: 4, name:'Tranfer'},
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.ts:20]: spelling mistake

File: src/app/application/dashboard/dashboard.component.ts [hunk 10]
```diff
@@ -1,18 +1,43 @@
...
+  this.modalService.open(AddAccountComponent, { centered: true, size: 'md', });
+}
+
+minToDate: any;
+maxDate={year:new Date().getFullYear(),month: new Date().getMonth()+1, day: new Date().getDate()};
+
+  public statementFiltersForm!: FormGroup;
+  constructor(private modalService: NgbModal,private formBuilder: FormBuilder,) { }
+  ngOnInit(): void {
+    this.model = { year: 2024, month: 4, day: 12 }
```
Comment (rachittyagi) [inline src/app/application/dashboard/dashboard.component.ts:41]: why hard coded date?

**Review by rachittyagi — CHANGES_REQUESTED** [review]

File: src/app/application/dashboard/dashboard.component.ts (hunk 9 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.ts]: changed

File: src/app/application/dashboard/dashboard.component.html (hunk 6 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: changed

File: src/app/application/dashboard/dashboard.component.html (hunk 4 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: done

File: src/app/application/dashboard/dashboard.component.html (hunk 3 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: done

File: src/app/application/dashboard/dashboard.component.css (hunk 2 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.css]: done

File: src/app/application/dashboard/dashboard.component.css (hunk 1 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.css]: done

File: src/app/application/dashboard/dashboard.component.html (hunk 7 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: done

File: src/app/application/dashboard/dashboard.component.html (hunk 4 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: attribute provides a label for the search input field, making it more accessible to users who rely on screen readers.

File: src/app/application/dashboard/dashboard.component.html (hunk 5 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html]: done

**Review by rachittyagi — APPROVED** [review]

**Pull request merged**